test/integration/camera_sensor.cc

```diff
@@ -716,138 +722,202 @@ TEST_F(CameraSensor, CheckDistortion)
   std::string cameraNameBarrel = "camera_sensor_barrel";
   std::string modelNamePincushion = "camera_model_pincushion";
   std::string cameraNamePincushion = "camera_sensor_pincushion";
+  std::string modelNameExtremeBarrel = "camera_model_extreme_barrel";
+  std::string cameraNameExtremeBarrel = "camera_sensor_extreme_barrel";
+  std::string modelNameExtremePincushion = "camera_model_extreme_pincushion";
+  std::string cameraNameExtremePincushion = "camera_sensor_extreme_pincushion";
   unsigned int width  = 320;
   unsigned int height = 240;
   double updateRate = 10;
   ignition::math::Pose3d setPose(
       ignition::math::Vector3d(-5, 0, 5),
       ignition::math::Quaterniond(0, IGN_DTOR(15), 0));
 
-  // spawn an undistorted camera
-  SpawnCamera(modelNameUndistorted, cameraNameUndistorted, setPose.Pos(),
-      setPose.Rot().Euler(), width, height, updateRate);
-  // spawn a flat camera
-  SpawnCamera(modelNameFlat, cameraNameFlat, setPose.Pos(),
-      setPose.Rot().Euler(), width, height, updateRate,
-      "", 0, 0, true, 0, 0, 0, 0, 0, 0.5, 0.5);
-  // spawn a camera with barrel distortion
-  SpawnCamera(modelNameBarrel, cameraNameBarrel, setPose.Pos(),
-      setPose.Rot().Euler(), width, height, updateRate,
-      "", 0, 0, true, -0.1349, -0.51868, -0.001, 0, 0, 0.5, 0.5);
-  // spawn a camera with pincushion distortion
-  SpawnCamera(modelNamePincushion, cameraNamePincushion, setPose.Pos(),
-      setPose.Rot().Euler(), width, height, updateRate,
-      "", 0, 0, true, 0.1349, 0.51868, 0.001, 0, 0, 0.5, 0.5);
-
-  sensors::SensorPtr sensorUndistorted =
-    sensors::get_sensor(cameraNameUndistorted);
-  sensors::CameraSensorPtr camSensorUndistorted =
-    std::dynamic_pointer_cast<sensors::CameraSensor>(sensorUndistorted);
-  sensors::SensorPtr sensorFlat =
-    sensors::get_sensor(cameraNameFlat);
-  sensors::CameraSensorPtr camSensorFlat =
-    std::dynamic_pointer_cast<sensors::CameraSensor>(sensorFlat);
-  sensors::SensorPtr sensorBarrel =
+  for (auto isUseLegacyDistortionMode: {true, false}) {
```

Comment:
The goal was to extend this test to cover the legacy and new modes. This was before I added the newest test. I didn't realize that they wouldn't spawn correctly. I can undo these changes, since I think the new distortion test is much more relevant.

Reply:
I removed these changes from the PR - the new test is still included.